pysnooper/tracer.py

```diff
@@ -19,10 +19,21 @@
 
 
 def get_local_reprs(frame, watch=()):
-    result = {key: utils.get_shortish_repr(value) for key, value
-                                                     in frame.f_locals.items()}
+    result_items = [(key, utils.get_shortish_repr(value)) for key, value
+                                                     in frame.f_locals.items()]
+
+    vars_order = frame.f_code.co_varnames + frame.f_code.co_cellvars + frame.f_code.co_freevars
```

Comment:
```suggestion
    vars_order = frame.f_code.co_varnames + frame.f_code.co_freevars
```

`co_cellvars` names variables in functions inside this one, we don't care about it.

Reply:
Here is an example why we should:

```python
import pysnooper

@pysnooper.snoop(depth=2)
def f1(a):
    def f2(a):
        x = 0
        x += 1
        def f3(a):
            y = x
            return 42
        return f3(a)
    return f2(a)

f1(1)
```

The x variable is in co_cellvars.
If we try sort with this simpler function, which ignores the unknown vars:
```python
result_items.sort(key=lambda key_value: vars_order.index(key_value[0]))
```
... we can get that is works if we use co_cellvars and doesn't work if we don't.

I think this simple function should always work on cpython, but I don't sure about other implementations.

Reply:
I checked out your repo and got the exact same results with your sample script with and without my change.

> If we try sort with this simpler function

but we're not, are we? I agree it's best to avoid relying too much on the workings of code objects.

Reply:
Your `vars_order` list is a great idea. How about you add f_locals to the end of it? Even if some variables repeat themselves, that wouldn't hurt. That way we'll be 100% sure no variable makes it out of the order, no matter what magic the user used when defining his function.

Reply:
And of course, that way we could remove this ugly check: ` if key_value[0] in vars_order else float("inf"))`

Reply:
Could you please repeat the test, because for me it is reproducible on Python 3.5, 3.6 3.7 and on PyPy/PyPy3.

The simpler function was only for example where co_cellvars considering is needed.

Reply:
> Your `vars_order` list is a great idea. How about you add f_locals to the end of it? Even if some variables repeat themselves, that wouldn't hurt. That way we'll be 100% sure no variable makes it out of the order, no matter what magic the user used when defining his function.

Done, https://github.com/cool-RR/PySnooper/pull/113/commits/00591f658859eb3c6c8fa33e0973bbb0368d0a06. The code in get_local_reprs became much nicer to read.